common/utils.go

```diff
@@ -73,3 +74,17 @@ func SleepWithContext(ctx context.Context, d time.Duration) error {
 		return nil
 	}
 }
+
+// GetFreeTCPPort returns an OS allocated and immediately freed port. There is
+// nothing preventing something else on the system from using the port before
+// the caller has a chance, but OS allocated ports are randomised to make the
+// risk negligible.
+func GetFreeTCPPort() (uint, error) {
```

Comment:
Previously this was a private function in the monero package (for picking a free monero-wallet-rpc port). Moved it here and made it public so others can use it.